index.js

```diff
@@ -89,11 +117,6 @@ function getOptions(uri, o, method) {
         o.timeout = 2 * 60 * 1000; // 2 minutes
     }
 
-    // Default pool options: Don't limit the number of sockets
-    if (!o.pool) {
-        o.pool = {maxSockets: Infinity};
-    }
-
```

Comment:
The commit msg for that change says,

> Setting any pool option will create a new agent of the httpModule
   class instead of using the global agent. So, the whole rigmarole with
   the connection timeout was never being used. Not sure what the
   desired behaviour is here, but it certainly isn't this.